sqlmesh/core/model/definition.py

```diff
@@ -919,8 +919,19 @@ def load_model(
 
     dialect = dialect or ""
     meta = expressions[0]
-    query = expressions[-1] if len(expressions) > 1 else None
-    statements = expressions[1:-1]
+
+    if any(isinstance(e, (exp.Subqueryable, d.Jinja)) for e in expressions):
+        try:
+            # Extract the query and any pre/post statements
+            query, pre, post = extract_sql_model_select(expressions[1:])
+        except Exception as e:
+            raise_config_error(f"Invalid model definition: {e}", path)
+        statements = pre + post
+    else:
+        # Assume this is a different model type (seed, python, etc.)
+        query = expressions[-1] if len(expressions) > 1 else None
```

Comment:
seed and python models don't have queries, why did you need this block?

Reply:
I fallback to the codepath that already existed because tests were getting messed up and it was really hard to figure out. The worst offender was the `test_no_query` or something.